### PR TITLE
Exposing events for the job so consumers can register their own event…

### PR DESCRIPTION
**Pull request by rpmoore**
… handlers.

### CODE REVIEW

File: ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/Smoke_Test.java [hunk 1]
```diff
@@ -688,4 +693,85 @@ public SeekableByteChannel buildChannel(final String key) throws IOException {
...
+                public SeekableByteChannel buildChannel(final String key) throws IOException {
+                    return new NullChannel();
+                }
+            });
+
+            assertThat(counter.get(), is(1));
+        } finally {
+            Util.deleteAllContents(client, bucketName);
+        }
+    }
```
Comment (DenverM80) [inline ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/Smoke_Test.java]: missing blank line between tests

File: ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/JobPartTracker.java [hunk 2]
```diff
@@ -50,4 +50,16 @@ public JobPartTracker attachObjectCompletedListener(final ObjectCompletedListene
         }
         return this;
     }
+
+    public void removeDataTransferredListener(final DataTransferredListener listener) {
+        for (final ObjectPartTracker tracker : this.trackers.values()) {
```
Comment (DenverM80) [inline ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/JobPartTracker.java]: Could a JobPartTracker have both a Data and Object tracker?  And if so, do we only want to iterate trackers of a certain type?  Not sure I'm following what's going on here clearly.

Comment (rpmoore) [inline ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/JobPartTracker.java]: A job can have many object part trackers.  The registration code does almost the same thing.

Comment (DenverM80) [inline ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/JobPartTracker.java]: My mistake, confused tracker with listener subtypes

**Pull request merged**